test/unit/host_test.rb

```diff
@@ -2174,6 +2174,13 @@ def to_managed!
     assert_equal 'foo', host.shortname
   end
 
+  test 'lookup_value_match returns host name instead of fqdn when there is no primary interface' do
+    host = FactoryGirl.build(:host, :managed)
+    host.interfaces.delete_all
+    assert_nil host.primary_interface
+    assert_equal host.send(:lookup_value_match), "fqdn=#{host.name}"
```

Comment:
In this instance, I'd consider specifying taking a copy of `host.name` before deleting the interface and using that here, or possibly even specifying :name in the factory.  Although it's initialised today, it may futureproof this test in case `host.name` becomes empty during interface deletion (e.g. the NameSynchronizer makes it blank).